src/socket/address.cr

```diff
@@ -107,8 +107,8 @@ class Socket
     def address
```

Comment:
@ysbaddaden @spalladino Please next time prefer adding a return type annotation here, which makes docs more clear and removes the need to write a `typeof` test for this, which is redundant